nilearn/plotting/img_plotting.py

```diff
@@ -1463,3 +1463,123 @@ def plot_connectome_strength(adjacency_matrix, node_coords, node_size="auto",
+                figure=None, axes=None, title=None):
+    """Plot an image representation of voxel intensities across time.
+
+    This figure is also known as a "grayplot" or "Power plot".
+
+    Parameters
+    ----------
+    img : Niimg-like object
+        4D image.
+        See http://nilearn.github.io/manipulating_images/input_output.html.
+    mask_img : Niimg-like object or None, optional
+        Limit plotted voxels to those inside the provided mask (default is
+        None). If not specified a new mask will be derived from data.
+        See http://nilearn.github.io/manipulating_images/input_output.html.
+    detrend : :obj:`bool`, optional
+        Detrend and z-score the data prior to plotting (default is `True`).
+    output_file : :obj:`str` or None, optional
+        The name of an image file to which to export the plot (default is
+        None). Valid extensions are .png, .pdf, and .svg.
+        If `output_file` is not None, the plot is saved to a file, and the
+        display is closed.
+    figure : :class:`matplotlib.figure.Figure` or None, optional
+        Matplotlib figure used (default is None).
+        If None is given, a new figure is created.
+    axes : :class:`matplotlib.axes._subplots.AxesSubplot` or None, optional
+        The axes used to display the plot (default is None).
+        If None, the complete figure is used.
+    title : :obj:`str` or None, optional
+        The title displayed on the figure (default is None).
+
+    Returns
+    -------
+    figure : :class:`matplotlib.figure.Figure`
+        Figure object with carpet plot.
+
+    Notes
+    -----
+    This figure was originally developed in [1]_.
+
+    References
+    ----------
+    .. [1] Power, J. D. (2017). A simple but useful way to assess fMRI scan
+    qualities. Neuroimage, 154, 150-158. doi:
+    https://doi.org/10.1016/j.neuroimage.2016.08.009
+    """
+    img = _utils.check_niimg_4d(img, dtype='auto')
+
+    # Define TR and number of frames
+    tr = img.header.get_zooms()[-1]
+    ntsteps = img.shape[-1]
+
+    if not mask_img:
+        mask_img = compute_epi_mask(img)
+    else:
+        mask_img = _utils.check_niimg_3d(mask_img, dtype='auto')
+
+    data = apply_mask(img, mask_img)
+    # Detrend and standardize data
+    if detrend:
+        data = clean(data, t_r=tr, detrend=True, standardize='zscore')
+
+    if not figure:
+        if not axes:
+            figure = plt.figure()
+        else:
+            figure = axes.figure
+
+    if not axes:
+        axes = figure.add_subplot(1, 1, 1)
+    else:
+        assert axes.figure is figure, ("The axes passed are not "
+                                       "in the figure")
+
+    # Avoid segmentation faults for long acquisitions by decimating the data
+    LONG_CUTOFF = 800
+    if ntsteps > LONG_CUTOFF:
+        data = data[::2, :]
+
+    axes.imshow(data.T, interpolation='nearest',
+                aspect='auto', cmap='gray', vmin=-2, vmax=2)
```

Comment:
The vmin and the vmax seem very arbitrary to me. Images could be scaled very differently.

Reply:
Do you want me to add them as arguments?

Reply:
I've added them as arguments and switched to calculating M +/- 2SD from the data, since the +/- 2 were good defaults for _z-scored_ data. This should work well when `detrend=False` as well.